18F/identity-idp
# LG-11883: Add validation to GpoConfirmation entries

## 🎫 Ticket

[LG-11883](https://cm-jira.usa.gov/browse/LG-11883)

## 🛠 Summary of changes

This PR tackles handling of invalid `GpoConfirmation` records in a couple of places:

**Closer to the user**

First, it adds validation to various required fields on `GpoConfirmation` to ensure that, before we write the record for the user, the required information is in place. If validation fails, an error is raised and the user sees an HTTP 500 error. At this point, the way forward for the user is to restart IdV (we may consider enhancing this if we see additional evidence of the underlying problem in prod).

**On export**

It also updates the daily job that uploads `GpoConfirmation` records to:

* Exclude any invalid records from the upload 
* Preserve invalid records in the database for investigation
* Notify NewRelic if any invalid records were encountered.

## 📜 Testing Plan

This PR adds a new config, `invalid_gpo_confirmation_zipcode`. If set, and you enter that zipcode when going through IdV, your GpoConfirmation record won't validate and you'll get to reproduce the bug.

1. Start IdV
2. Edit your address to use the zip code "00001"
3. Enter the verify by mail flow
4. Enter your password
5. Get an error screen after entering your password
